nn_sdwan/tests/test_nn_sdwan.py

```diff
@@ -0,0 +1,58 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import pytest
+
+from datadog_checks.base import ConfigurationError
+from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.nn_sdwan import NnSdwanCheck
+
+from .metrics import ALL_METRICS
+
+
+@pytest.mark.unit
+def test_empty_instance(aggregator, instance_empty):
+    with pytest.raises(ConfigurationError):
+        NnSdwanCheck('nn_sdwan', {}, [instance_empty])
+
+
+@pytest.mark.unit
+def test_missing_hostname(aggregator, instance_missing_hostname):
+    with pytest.raises(ConfigurationError):
+        NnSdwanCheck('nn_sdwan', {}, [instance_missing_hostname])
+
+
+@pytest.mark.unit
+def test_missing_username(aggregator, instance_missing_username):
+    with pytest.raises(ConfigurationError):
+        NnSdwanCheck('nn_sdwan', {}, [instance_missing_username])
+
+
+@pytest.mark.unit
+def test_missing_password(aggregator, instance_missing_password):
+    with pytest.raises(ConfigurationError):
+        NnSdwanCheck('nn_sdwan', {}, [instance_missing_password])
+
+
+@pytest.mark.unit
+def test_missing_protocol(aggregator, instance_missing_protocol):
+    with pytest.raises(ConfigurationError):
+        NnSdwanCheck('nn_sdwan', {}, [instance_missing_protocol])
+
+
+@pytest.mark.unit
+def test_normal_instance(aggregator, instance_normal):
+    NnSdwanCheck('nn_sdwan', {}, [instance_normal])
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_metrics_received(aggregator, instance_normal):
+    check = NnSdwanCheck('nn_sdwan', {}, [instance_normal])
+
+    check.check(instance_normal)
+    for metric in ALL_METRICS:
+        aggregator.assert_metric(metric, at_least=0)
+
+    aggregator.assert_all_metrics_covered()
```

Comment:
This test is failing in the CI:
```

E     Asserted Metrics:
E     	- nn_sdwan.app_aware_routing.jitter
E     	- nn_sdwan.app_aware_routing.latency
E     	- nn_sdwan.app_aware_routing.loss_percentage
E     	- nn_sdwan.app_aware_routing.rx_octets
E     	- nn_sdwan.app_aware_routing.tx_octets
E     	- nn_sdwan.cert_summary
E     	- nn_sdwan.connection_summary_stats_error
E     	- nn_sdwan.connection_summary_stats_total
E     	- nn_sdwan.device_control_status
E     	- nn_sdwan.reboot_count
E     	- nn_sdwan.site_health
E     	- nn_sdwan.top_app_stats
E     	- nn_sdwan.transport_interface
E     	- nn_sdwan.vmanage_count
E     	- nn_sdwan.wan_edge_health
E     	- nn_sdwan.wan_edge_inventory
E     Found metrics that are not asserted:
E     	- 
```

These metrics do not appear to be emitted in the test, so the test is failing since they are asserted but not emitted in `assert_all_metrics_covered`. Typically a docker environment or test fixtures will need to be added to properly collect the metrics in tests. You can either look into this, or stop asserting metrics (which will decrease code coverage).

Reply:
Unfortunately, we can't put our required environment (SD-WAN controllers) in a docker environment, so I suppose we'll stop asserting the metrics.